blocks/library/heading/index.js

```diff
@@ -38,9 +43,9 @@ registerBlock( 'core/heading', {
 				blocks: [ 'core/text' ],
 				transform: ( { content, align } ) => {
 					if ( Array.isArray( content ) ) {
-						// TODO this appears to always be true?
-						// TODO reject the switch if more than one paragraph
-						content = content[ 0 ];
+						content = wp.element.concatChildren( content.map( ( elt ) =>
+							! elt || isString( elt ) || elt.type !== 'p' ? elt : elt.props.children
```

Comment:
Quite unfortunate that the heading transform needs to be aware for text forms as an array, as a falsey value, as a string, as an element of paragraph type, or other element type 😕